api-server/src/uploads/uploads.controller.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Controller,
+  FileTypeValidator,
+  HttpStatus,
+  MaxFileSizeValidator,
+  ParseFilePipe,
+  ParseFilePipeBuilder,
+  Post,
+  Req,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { VERSION_1_URI } from '../utils/versionts';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { EventsResponse } from '../events/dto/events-response';
+import { UploadsResponse } from './dto/UploadsResponse';
+import { UploadsService } from './uploads.service';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Request } from 'express';
+
+const TEN_MEGABYTES = 10000000;
+
+@Controller(`${VERSION_1_URI}/uploads`)
+@ApiTags('uploads')
+export class UploadsController {
+  constructor(private readonly uploadsService: UploadsService) {}
+
+  @Post('event-image')
+  @ApiOperation({
+    summary: 'uploads an image for reference by events',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'the path to the image on the server',
+    type: EventsResponse,
+  })
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadEventImage(
+    @UploadedFile(getImageUploadValidators())
+    file: Express.Multer.File,
+    @Req() request: Request,
+  ): Promise<UploadsResponse> {
+    const imagePath = await this.uploadsService.saveImage(file);
+    return new UploadsResponse({ imagePath });
+  }
+}
+
+function getImageUploadValidators(): ParseFilePipe {
+  return new ParseFilePipeBuilder()
+    .addFileTypeValidator({
+      fileType: /image\/(jpe?g|png|webp|tif|tiff|bmp)$/i,
```

Comment:
Does this support `image/*`? That's what the input on the form uses for `accept` (per [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers) it should accept any image format).

(I wish Nestjs had API docs to supplement the higher-level ones.)

Reply:
yeah this was quite frustrating to get right what I'm matching on are mime types such as image/jpeg, image/png. It's just a regular expression so you could match against image/*, that would be `/image\/.*$/i` I believe.

If that's what we're allowing in the form that would make sense though long term we should probably lock this down more. I doubt we want to support any image format

Reply:
loosened up the requirements 9bf391a22a4a612f87b00a114fc108ff9f614b97